NoroffFEU/workflow-repo-ca
# WORKFLOW CA: Add unit, e2e tests and readme to project

### Overview: 

This PR adds 2 unit tests to the isActivePath and getByUserName functions as well as 2 e2e playwright tests to test login and navigation functionality. 

### Motivation:

Apply the skills and knowledge gained during the workflow course to enhance the quality of a website, done by configuring tools and defining tests that improve the efficiency of the development process.

### Changes:

- Configured Vite, Eslint, Prettier, Husky.
- Configured Vitest and Playwright for testing. 
- Added unit tests for isActivePath and getByUserName functions.
- Added e2e tests for login and navigation functionality. 
- Added environment variables and .env.example.
- Updated a README.md file.

### Testing: 

1. Make sure to correctly install all dependencies and packages. 
2. Edit .env.example or make a .env and insert credentials. 
3. Unit tests: `npm run test`.
4. E2E tests: `npx playwright test.`
5. Ensure all tests pass.
6. Give feedback / improvements. 

### Checklist: 

- [x] Eslint is installed and configured to handle test globals
- [x] Prettier is installed and configured
- [x] Pre-commit hooks have been set up to handle linting and formatting
- [x] Vitest has been installed and configured
- [x] The relevant unit tests have been written and pass
- [x] Playwright has been installed and configured
- [x] The relevant e2e tests have been written and pass
- [x] The README has been updated with the relevant scripts and environment variable examples
- [x] .env is in .gitignore
- [x] There is a .env.example file in the branch

Teacher review: @cnnrbrn
